### ISSUE TITLE
Reshaping error when solving problem

### ISSUE THREAD

**Issue by alanhdu** [cvxgrp/cvxpy#530]
This is on a Ubuntu 16.04 computer on Python 3.6.5 with

```
cvxpy==1.0.6
cvxopt==1.2.0
numpy==1.14.5
```

the following script:

```python
import cvxpy
import numpy as np

common = cvxpy.Variable([16, 16], PSD=True)

x, y = np.random.rand(16, 16), np.random.rand(16, 16)
x = x @ x.T   # make PSD
y = y @ y.T

diff_x, diff_y = x - common, y - common

constraints = [diff_x >> 0, diff_y >> 0]
loss = cvxpy.sum_squares(diff_x) + cvxpy.sum_squares(diff_y)

cvxpy.Problem(cvxpy.Minimize(loss), constraints).solve(solver=cvxpy.CVXOPT)
```

I often get an reshaping error in cvxpy:
```python-traceback
Traceback (most recent call last):
  File "test.py", line 15, in <module>
    cvxpy.Problem(cvxpy.Minimize(loss), constraints).solve(solver=cvxpy.CVXOPT)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/cvxpy/problems/problem.py", line 246, in solve
    return solve_func(self, *args, **kwargs)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/cvxpy/problems/problem.py", line 361, in _solve
    self.unpack_results(solution, self._solving_chain, inverse_data)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/cvxpy/problems/problem.py", line 453, in unpack_results
    solution = chain.invert(solution, inverse_data)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/cvxpy/reductions/chain.py", line 66, in invert
    solution = r.invert(solution, inv)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/cvxpy/reductions/matrix_stuffing.py", line 114, in invert
    order='F'
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 257, in reshape
    return _wrapfunc(a, 'reshape', newshape, order=order)
  File "/home/alan/workspace/ctrl-labs/py3.venv/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 52, in _wrapfunc
    return getattr(obj, method)(*args, **kwds)
ValueError: cannot reshape array of size 255 into shape (16,16)
```

This seems to be a something introduced between `cvxpy==1.0.4` (which runs fine) and `cvxpy==1.0.5` (which crashes).

**Comment by rileyjmurray** [comment]
This error seems similar to #510, which uses CVXOPT and cvxpy 1.0.3. Can you try the same problem with SCS and cvxpy 1.0.5?

**Comment by alanhdu** [comment]
Sorry for the delay! I re-ran this script (swapping out `solver=cvxpy.CVXOPT` with `solver=cvxpy.SCS`) using `cvxpy==1.0.5` and `scs==2.0.2` and this seems to have run without errors (although it was a little slow on my laptop).

**Comment by rileyjmurray** [comment]
I just ran the script on a modified version of cvxpy (containing a fix for #510), and CVXOPT worked without error. I also checked optimality conditions for the dual variables, and they seemed to work out fine. I'm marking this as "will be fixed soon".
